src/openrct2/ride/Vehicle.cpp

```diff
@@ -6600,26 +6600,34 @@ static void vehicle_update_track_motion_up_stop_check(rct_vehicle * vehicle)
 }
 
 /**
- * Modifies the train's velocity to match the block-brake fixed velocity.
+ * Modifies the train's velocity to match the block-brakes specified velocity.
  * This function must be called when the car is running through a non-stopping
  * state block-brake (precondition), which means that the block brake is acting
  * merely as a velocity regulator, in a closed state. When the brake is open, it
- * boosts the train to the speed limit
+ * boosts the train to the fixed block release speed.
  */
-static void apply_non_stop_block_brake(rct_vehicle * vehicle, bool block_brake_closed)
+static void apply_non_stop_block_brake(rct_vehicle * vehicle, bool block_brake_closed, int speed)
 {
+    if (speed < 0x20364)
+    {
+        //Fix blocks that were saved before block section speed settings were implemented.
+        speed = 0x20364;
+    }
     if (vehicle->velocity >= 0)
     {
         // If the vehicle is below the speed limit
-        if (vehicle->velocity <= 0x20364)
+        if (vehicle->velocity <= speed)
         {
-            // Boost it to the fixed block brake speed
-            vehicle->velocity     = 0x20364;
-            vehicle->acceleration = 0;
+            if (vehicle->velocity <= 0x20364)
```

Comment:
Could you create a define for this? `#define BLOCK_BRAKE_BASE_SPEED 0x20364`

Reply:
`constexpr auto BLOCK_BRAKE_BASE_SPEED = 0x20364;`

Reply:
Well, yeah, go with @IntelOrca 's suggestion.

Reply:
Done